pkg/apis/fake/v1alpha1/db_types.go

```diff
@@ -25,27 +25,52 @@ import (
 // DBSpec defines the desired state of DB
 type DBSpec struct {
 	// RootPassword is root account password for this data. Leave blank for auto-generation
-	RootPassword string `json:"rootPassword"`
+	RootPassword string `json:"rootPassword,omitempty"`
 
 	// DatabaseName of the created database at start up
+	// +kubebuilder:validation:MinLength=1
 	DatabaseName string `json:"databaseName"`
 
 	// DatabaseUser for created database
 	DatabaseUser string `json:"databaseUser"`
 
 	// DatabasePassword for the created database. Leave blank for auto-generation
-	DatabasePassword string `json:"databasePassword"`
+	DatabasePassword string `json:"databasePassword,omitempty"`
 
-	// E2E tests params
+	// Config is E2E tests params
 	Config Config `json:"config,omitempty"`
 }
 
+// OperationFlagType represents a enable/disable flag
+type OperationFlagType string
+
+func (o OperationFlagType) Enabled() bool {
+	return o == OperationFlagEnabled
+}
+
+// Values of OperationFlagType.
+const (
+	OperationFlagEnabled  OperationFlagType = "enabled"
+	OperationFlagDisabled OperationFlagType = "disabled"
+)
+
 // Config defines the e2e tests params
 type Config struct {
 	// ReadyAfterSeconds represents duration after which operator will mark DB as Ready
+	// +kubebuilder:default:=0
```

Comment:
why do we need the `default:=0`, it should be 0 by default?

Reply:
you're right